src/value/number.rs

```diff
@@ -0,0 +1,473 @@
+    /// assert_eq!(v["b"].as_f64(), Some(64.0));
+    /// assert_eq!(v["c"].as_f64(), Some(-64.0));
+    /// # }
+    /// ```
+    ///
+    /// ```rust
+    /// # use std::f64;
+    /// # fn yaml(i: &str) -> serde_yaml::Value { serde_yaml::from_str(i).unwrap() }
+    /// assert_eq!(yaml("inf").as_f64(), Some(f64::INFINITY));
+    /// assert_eq!(yaml("-inf").as_f64(), Some(f64::NEG_INFINITY));
+    /// assert!(yaml("NaN").as_f64().unwrap().is_nan());
+    /// ```
+    #[inline]
+    pub fn as_f64(&self) -> Option<f64> {
+        match self.n {
+            N::PosInt(n) => NumCast::from(n),
+            N::NegInt(n) => NumCast::from(n),
+            N::Float(n) => Some(n),
+        }
+    }
+
+    /// Converts a finite `f64` to a `Number`. Infinite or NaN values are YAML
+    /// numbers.
+    ///
+    /// ```rust
+    /// # use std::f64;
+    /// #
+    /// # use serde_yaml::Number;
+    /// #
+    /// assert!(Number::from_f64(256.0).is_finite());
+    ///
+    /// assert!(Number::from_f64(f64::NAN).is_nan());
+    ///
+    /// assert!(!Number::from_f64(f64::INFINITY).is_finite());
+    ///
+    /// assert!(!Number::from_f64(f64::NEG_INFINITY).is_finite());
+    /// ```
+    #[inline]
+    pub fn from_f64(f: f64) -> Number {
+        Number { n: N::Float(f) }
+    }
+
+    /// Checks to see if a number is finite or not
+    ///
+    /// ```rust
+    /// # use std::f64;
+    /// #
+    /// # use serde_yaml::Number;
+    /// #
+    /// assert!(Number::from_f64(256.0).is_finite());
+    ///
+    /// assert!(!Number::from_f64(f64::NAN).is_finite());
+    ///
+    /// assert!(!Number::from_f64(f64::INFINITY).is_finite());
+    ///
+    /// assert!(!Number::from_f64(f64::NEG_INFINITY).is_finite());
+    ///
+    /// assert!(Number::from(1).is_finite());
+    /// ```
+    #[inline]
+    pub fn is_finite(&self) -> bool {
+        match self.n {
+            N::PosInt(_) | N::NegInt(_) => true,
+            N::Float(f) => f.is_finite(),
+        }
+    }
+
+    /// Checks to see if a number is a number
+    ///
+    /// ```rust
+    /// # use std::f64;
+    /// #
+    /// # use serde_yaml::Number;
+    /// #
+    /// assert!(!Number::from_f64(256.0).is_nan());
+    ///
+    /// assert!(Number::from_f64(f64::NAN).is_nan());
+    ///
+    /// assert!(!Number::from_f64(f64::INFINITY).is_nan());
+    ///
+    /// assert!(!Number::from_f64(f64::NEG_INFINITY).is_nan());
+    ///
+    /// assert!(!Number::from(1).is_nan());
+    /// ```
+    #[inline]
+    pub fn is_nan(&self) -> bool {
+        match self.n {
+            N::PosInt(_) | N::NegInt(_) => false,
+            N::Float(f) => f.is_nan(),
+        }
+    }
+}
+
+impl fmt::Display for Number {
+    fn fmt(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+        match self.n {
+            N::PosInt(i) => Display::fmt(&i, formatter),
+            N::NegInt(i) => Display::fmt(&i, formatter),
+            N::Float(f) => Display::fmt(&f, formatter),
```

Comment:
This should have a special case for inf and nan. YAML has a distinct formatting for them.